### ISSUE TITLE
Arduino Zero / MKR1000 : first byte of clockless pixel data not sent

### ISSUE THREAD

**Issue by Daniel-dk** [FastLED/FastLED#278]
This *might* be related to issue #269 ( missing byte of data )
TLDR: it doesn't look like the first byte is being sent with clockless controller
using WS2812B LEDs with Arduino Zero / MKR1000 ( ATSAMD21G18A )
- the pin/port mapping is different but does the same on both controllers/bootloaders 
- I added the MKR1000 pin map to fastpin_arm_d21.h
- Timing also seems a bit strange but I am unsure where to change it only for this platform - tweaking in \FastLED\platforms\arm\common\m0clockless.h ? but Assembler is voodoo to me
- Data doesn't seem to go missing when I use WS2801 ( SPI like ) controller, only for the clockless WS2812B .
- The first byte is stil missing on any other pixels sent ( NUM_PIXELS = 2, and pixel[1] = CRGB(255,255,255);

attached are Logic analyser (saleae logic) captures of the following code:

```
// MKR1000
#define DATA_PIN 15 // 14 on Arduino Zero
#define LOGIC_PIN 6 // 6 on Arduino Zero
// How many leds in your strip?
#define NUM_LEDS 1
// Define the array of leds
CRGB leds[NUM_LEDS];

void setup() {
        // - keep RGB mapping as "RGB" so we know which byte is missing
	FastLED.addLeds<WS2812B, DATA_PIN,RGB>(leds, NUM_LEDS); // WS2812B pixel
	FastLED.addLeds<WS2812B, LOGIC_PIN, RGB>(leds, NUM_LEDS); // Logic analyser

// - so Fastled doesn't correct our output
	FastLED.setBrightness(255);
	FastLED.setCorrection(UncorrectedColor);
	FastLED.setDither(DISABLE_DITHER);
}

void loop() {
	leds[0] = CRGB(255,255,255); // White
	FastLED.show();
	leds[0] = CRGB(255, 0, 0);  // red/green depending on order
	FastLED.show();
	leds[0] = CRGB(0, 255, 0);  // red/green depending on order
	FastLED.show();
	leds[0] = CRGB(0, 0, 255); // blue
	FastLED.show();
	delay(200);
}
```
using `FastLED.addLeds<WS2812B, DATA_PIN,RGB>(leds, NUM_LEDS); // WS2812B pixel`
`leds[0] = CRGB(255,255,255); // White`
![255_255_255](https://cloud.githubusercontent.com/assets/582384/14049065/3f8be190-f2bb-11e5-89dc-efe90f8ca929.PNG)

`leds[0] = CRGB(255, 0, 0);  // red/green depending on order`
![255_0_0](https://cloud.githubusercontent.com/assets/582384/14049066/3f931636-f2bb-11e5-9143-ffad2aaf16c8.PNG)

`leds[0] = CRGB(0, 255, 0);  // red/green depending on order`
![0_255_0](https://cloud.githubusercontent.com/assets/582384/14049064/3f874ac2-f2bb-11e5-901e-8bd11e32e619.PNG)

`leds[0] = CRGB(0, 0, 255); // blue`
![0_0_255](https://cloud.githubusercontent.com/assets/582384/14049063/3f83a6c4-f2bb-11e5-912d-b87a9747344a.PNG)

[saleae logic 1.2.5 ](https://www.saleae.com/downloads)capture files : [capture.zip](https://github.com/FastLED/FastLED/files/189775/capture.zip)

When sending more than 1 pixel the first byte is still missing :
```
        leds[0] = CRGB(255,255,255); // White
	leds[1] = CRGB(255, 255, 255); // White
	FastLED.show();
```
![2pixel_255_255_255](https://cloud.githubusercontent.com/assets/582384/14049932/06042fda-f2c1-11e5-898c-d059450090bd.PNG)

I added the MKR1000 pin mapping to fastpin_arm_d21.h after the ARDUINO_SAMD_WINO's #elif
```
#elif defined(ARDUINO_SAMD_MKR1000)

#define MAX_PIN 22
_DEFPIN_ARM(  0, 0, 22); _DEFPIN_ARM(  1, 0, 23); _DEFPIN_ARM(  2, 0, 10); _DEFPIN_ARM(  3, 0, 11);
_DEFPIN_ARM(  4, 1, 10); _DEFPIN_ARM(  5, 1, 11); _DEFPIN_ARM(  6, 0, 20); _DEFPIN_ARM(  7, 0, 21);
_DEFPIN_ARM(  8, 0, 16); _DEFPIN_ARM(  9, 0, 17); _DEFPIN_ARM( 10, 0, 19); _DEFPIN_ARM( 11, 0,  8);
_DEFPIN_ARM( 12, 0,  9); _DEFPIN_ARM( 13, 1, 23); _DEFPIN_ARM( 14, 1, 22); _DEFPIN_ARM( 15, 0,  2);
_DEFPIN_ARM( 16, 1,  2); _DEFPIN_ARM( 17, 1,  3); _DEFPIN_ARM( 18, 0,  4); _DEFPIN_ARM( 19, 0,  5);
_DEFPIN_ARM( 20, 0,  6); _DEFPIN_ARM( 21, 0,  7);

#define SPI_DATA 8
#define SPI_CLOCK 9

#define HAS_HARDWARE_PIN_SUPPORT 1
```
as a comparison, this is what the WS2801 ( SPI ) output looks like - no missing byte
`leds[0] = CRGB(255,255,255); // White`
![ws2801_255_255_255](https://cloud.githubusercontent.com/assets/582384/14049744/e19ac088-f2bf-11e5-84fa-bc6f845e57c4.PNG)

leds[0] = CRGB(255, 0, 0);  // red/green depending on order
![ws2801_255_0_0](https://cloud.githubusercontent.com/assets/582384/14049743/e195879e-f2bf-11e5-8300-93b0659db423.PNG)

leds[0] = CRGB(0, 255, 0);  // red/green depending on order
![ws2801_0_255_0](https://cloud.githubusercontent.com/assets/582384/14049742/e18f7840-f2bf-11e5-9fcc-04f19d11a40e.PNG)

leds[0] = CRGB(0, 0, 255); // blue
![ws2801_0_0_255](https://cloud.githubusercontent.com/assets/582384/14049745/e1a213e2-f2bf-11e5-9ed9-edd7a0f24883.PNG)

```

**Comment by focalintent** [comment]
What does it do for the second led - e.g. is it 0'ing out the first byte for all leds or just the first one - I don't have an m0 based setup here to test with right now, and won't be able to until later this weekend?

(#269 is a slightly different issue - or at least likely different cause - the code between avr and m0 arm is, to put it mildly, wildly different).

**Comment by Daniel-dk** [comment]
The first byte of led[1] is also 0-ing out :  ![pixel[1](255,255,255)](https://cloud.githubusercontent.com/assets/582384/14050112/5ae9f330-f2c2-11e5-8d1d-ade8a7684b8e.PNG)

**Comment by focalintent** [comment]
Ah!  I see the problem w/the m0 code - i'll try checking in a fix shortly..

**Comment by Daniel-dk** [comment]
Thank you ( We're building some MKR1000 based controllers- will send you one )

**Comment by focalintent** [comment]
Try pulling master@HEAD and see if that fixes it.

**Comment by Daniel-dk** [comment]
Works perfectly now, Thank you 
:
`leds[0] = CRGB(255,255,255);`
![255_255_255](https://cloud.githubusercontent.com/assets/582384/14050477/bf041344-f2c4-11e5-905b-8a24379a1958.PNG)
`leds[0] = CRGB(255,0,0); `
![255_0_0](https://cloud.githubusercontent.com/assets/582384/14050478/c231a306-f2c4-11e5-9ee2-211fa9580269.PNG)
